### PR TITLE
CONTRAST-29476: initial checkin of new .NET Core agent docs. 

### PR DESCRIPTION
**Pull request by ldemidov**
WIP - do not merge.

This PR is here so we can have conversation threads about the content of these new docs.

### CODE REVIEW

File: content/installation/net/configuration/YAML-properties-net.md [hunk 1]
```diff
@@ -61,6 +61,10 @@ All properties in this section must be put under the `agent` node, as shown in t
 
 * **agent**:
 
+#### Auto-update
+
+Use the following properties to control auto-update behavior of the .NET Framework Agent.  These settings do not apply to the .NET Core agent, which does not have the capability to auto-update.
+
```
Comment (kaitlynmortimer) [inline content/installation/net/configuration/YAML-properties-net.md]: Is the original .NET agent officially changing names to ".NET Framework"?

File: content/installation/netcore/usage/UsingWithIISExpress.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<!--
+title: "Using Contrast With IIS Express"
+description: "Guide to using IIS Express"
+tags: "installation tray configuration IIS express agent .Net"
+-->
+
```
Comment (kaitlynmortimer) [inline content/installation/netcore/usage/UsingWithIISExpress.md]: If no more content will be added here, we should add this bit to the General Usage article, and delete this one.

File: content/installation/netcore/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  * Supported Technologies
+    - /netcore/install/Supported-technology-netcore.md
+  * Installation 
+    - /netcore/install/NetCoreInstallation.md
+  * Usage:         
+    - /netcore/usage/GeneralUsage.md
+    - /netcore/usage/UsingDotNetAzure.md
+    - /netcore/usage/UsingEithIISExpress.md
+  * Configuration:
+    - Same as .NET: link to the same page
```
Comment (kaitlynmortimer) [inline content/installation/netcore/README.md]: Does this include the General Settings, Application-Specific Settings and Settings for Azure App Service articles for the .NET agent?

Comment (ldemidov) [reply content/installation/netcore/README.md]: No - just the YAML Properties and YAML Template section.  I guess this should be a separate doc then?

File: content/installation/net/configuration/YAML-properties-net.md (hunk 1 above)
Comment (ldemidov) [reply content/installation/net/configuration/YAML-properties-net.md]: Well don't know about "official name change" but I think we shouldn't say just say ".NET Agent" in the document text anymore.  We should be more specific to be less ambiguous.  Maybe 

I would suggest we only have the following 3 names:

.NET Framework Agent - when referring only to the current agent
.NET Core Agent - when referring only to the upcoming .NET Core agent
.NET Agents - when referring to both or either agent

@DanFiedler can chime in here.

File: content/installation/netcore/usage/UsingWithIISExpress.md (hunk 2 above)
Comment (ldemidov) [reply content/installation/netcore/usage/UsingWithIISExpress.md]: Yea, ok with deleting this for now.  We will support eventually IIS Express in NetCore but not in the initial release.  The installation instructions would be different from the NetFramework agent.

File: content/installation/netcore/usage/GeneralUsage.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+The agent automatically starts with your application as long as the environment is setup as described in [.NET Core installation](installation-netcoreinstall.html).  
+
+### Stop the agent 
+
+To stop the agent, stop the application and remove agent from its environment. Alternately, you may change the `CORECLR_ENABLE_PROFILING` setting to `0`.
+
+## IIS Express 
+
+The installation for the IIS Express users is the same as self-hosted applications.
```
Comment (kaitlynmortimer) [inline content/installation/netcore/usage/GeneralUsage.md]: Can we move this IIS section up to the Installation article?

File: content/installation/netcore/usage/UsingDotNetAzure.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<!--
+title: "Using Contrast .NET on Microsoft Azure"
+description: "Using Contrast .NET on Microsoft Azure"
+tags: "configuration microsoft azure cloud agent installation .Net"
+-->
+
+Use the Contrast .NET agent to analyze ASP.NET applications running on Azure Virtual Machines (VMs), Cloud Services, Mobile Services or Azure App Service (formerly Azure Web Sites). Follow the instructions below to set up the .NET agent in your environment.
```
Comment (kaitlynmortimer) [inline content/installation/netcore/usage/UsingDotNetAzure.md]: Have you guys decided when you're adding Azure content? If we're not planning on it for the first publication, I'll remove this article from the page structure.

File: content/installation/netcore/install/System-requirements-netcore.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+- You have administrative access to a web server, and the server is supported by Contrast.
+- There is a deployed application to be analyzed, and the web application technology is supported by Contrast.
+- The web server has network connectivity with Contrast. 
+- The server meets the minimum requirements (stated below). 
+
+## Minimum Requirements
+
+* .NET Framework >= 4.7.0 installed
+* .NET Core >= 2.1 < 3.0 (installed system-wide)
```
Comment (kaitlynmortimer) [inline content/installation/netcore/install/System-requirements-netcore.md]: Is this supposed to mean "greater than or equal to"?

File: content/installation/netcore/usage/UsingDotNetAzure.md (hunk 5 above)
Comment (Silvenga) [reply content/installation/netcore/usage/UsingDotNetAzure.md]: We will be supporting netcore on things like Azure Virtual Machines, but not the majority of Azure netcore deployments, e.g. Azure App Service.

@DanFiedler

File: content/installation/netcore/install/System-requirements-netcore.md (hunk 6 above)
Comment (Silvenga) [reply content/installation/netcore/install/System-requirements-netcore.md]: Yep.

File: content/installation/netcore/usage/GeneralUsage.md (hunk 4 above)
Comment (Silvenga) [reply content/installation/netcore/usage/GeneralUsage.md]: Yeah, that would probably be better.

File: content/installation/netcore/usage/UsingDotNetAzure.md (hunk 5 above)
Comment (kaitlynmortimer) [reply content/installation/netcore/usage/UsingDotNetAzure.md]: @ldemidov or @DanFiedler ?

Comment (ldemidov) [reply content/installation/netcore/usage/UsingDotNetAzure.md]: No Azure content for NetCore right now.

File: content/installation/netcore/install/Supported-technology-netcore.md [hunk 7]
```diff
@@ -25,6 +25,70 @@ The Contrast .NET Core agent supports analysis of web applications built on the
 > * .NET Core >= 3.0 is currently not supported.
 > * ASP.NET Core applications running under the .NET framework aren't supported.
 
-<!-- ## .NET Core for Azure App Service 
+<!-- ## .NET Core Rule support.  This could be in its own section -->
 
-TBD if we will support in the first release. -->
\ No newline at end of file
+## .NET Core Agent Rule Support
```
Comment (ldemidov) [inline content/installation/netcore/install/Supported-technology-netcore.md]: @DanFiedler I added this section for currently supported and upcoming.  Is this the best place to put or should it go in its own page?

File: content/installation/netcore/install/Rules-netcore.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
+
+<!--
+title: "Contrast .NET Core Agent Rules Support"
+description: "Assess and Protect rules supported by the Contrast .NET Core agent"
+tags: "installation agent .NET Core windows assess protect rules"
+-->
+
+
```
Comment (kaitlynmortimer) [inline content/installation/netcore/install/Rules-netcore.md]: @ldemidov and @DanFiedler 
I moved these to their own article. We can keep it on the page with Supported Technologies, as it was, or create a new Policy page to mirror the .NET docs. Either way is fine by me.

File: content/installation/netcore/install/System-requirements-netcore.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+- You have administrative access to a web server, and the server is supported by Contrast.
+- There is a deployed application to be analyzed, and the web application technology is supported by Contrast.
+- The web server has network connectivity with Contrast. 
+- The server meets the minimum requirements (stated below). 
+
+## Minimum Requirements
+
+* .NET Framework version 4.7.0 or above installed
+* .NET Core versions 2.1 or 2.2 (installed system wide)
+* 2 CPUs
```
[the remark this answers is not in the document]
Comment (Silvenga) [reply content/installation/netcore/install/System-requirements-netcore.md]: We currently don't support anything below 2.1 e.g. 2.0 or 1.1. So `>= 2.1 < 3.0` would be most correct.

File: content/installation/netcore/install/Rules-netcore.md (hunk 8 above)
Comment (ldemidov) [reply content/installation/netcore/install/Rules-netcore.md]: I think keeping it on Supported Technologies is ok for now.

I think we don't need a Policy page for netcore yet.  It's not tested and we may want to pare that page down to just file analysis rules anyway.  I'm curious what Dan thinks about that.

**Pull request merged**